src/encord_active/lib/metrics/acquisition_functions.py

```diff
@@ -0,0 +1,169 @@
+from abc import abstractmethod
+from pathlib import Path
+from typing import Any, Optional, Union
+
+import numpy as np
+from PIL import Image
+
+from encord_active.lib.common.iterator import Iterator
+from encord_active.lib.labels.classification import ClassificationType
+from encord_active.lib.labels.object import ObjectShape
+from encord_active.lib.metrics.metric import (
+    AnnotationType,
+    DataType,
+    EmbeddingType,
+    Metric,
+    MetricType,
+)
+from encord_active.lib.metrics.writer import CSVMetricWriter
+
+
+class AcquisitionFunction(Metric):
+    def __init__(
+        self,
+        title: str,
+        short_description: str,
+        long_description: str,
+        metric_type: MetricType,
+        data_type: DataType,
+        annotation_type: list[Union[ObjectShape, ClassificationType]] = [],
+        embedding_type: Optional[EmbeddingType] = None,
+        model=None,
+    ):
+        self._model = model
+        super().__init__(
+            title, short_description, long_description, metric_type, data_type, annotation_type, embedding_type
+        )
+
+    def execute(self, iterator: Iterator, writer: CSVMetricWriter):
+        for _, img_pth in iterator.iterate(desc=f"Running {self.metadata.title} acquisition function"):
+            if img_pth is None:
+                continue
+            img = self.read_image(img_pth)
+            if img is None:
+                continue
+            pred_probs = self.get_predicted_class_probabilities(img)
+            if pred_probs is None:
+                continue
+            score = self.score_predicted_class_probabilities(pred_probs)
+            writer.write(score)
+
+    def get_predicted_class_probabilities(self, image) -> Optional[np.ndarray]:
+        image_array = np.asarray(image).flatten()
+        return self._model.predict_proba([image_array])
```

Comment:
I think checking the model probabilities should be performed in this function, just after feeding image to the model

Reply:
I followed your advise and put it there (now it lives in the ModelWrapper class).